docs/iris/src/developers_guide/graphics_tests.rst

```diff
@@ -38,80 +39,110 @@ Testing actual plot results introduces some significant difficulties :
-following changes in Iris or any of its dependencies, this is the process
-you now need to follow:
-
-#. Create a directory in iris/lib/iris/tests called 'result_image_comparison'.
-#. From your Iris root directory, run the tests by using the command: 
-   ``python setup.py test``.
-#. Navigate to iris/lib/iris/tests and run the command: ``python idiff.py``.
-   This will open a window for you to visually inspect the changes to the
-   graphic and then either accept or reject the new result.
-#. Upon acceptance of a change or a new image, a copy of the output PNG file
-   is added to the reference image repository in
-   https://github.com/SciTools/test-images-scitools.  The file is named
-   according to the image hash value, as ``<hash>.png``.
-#. The hash value of the new result is added into the relevant set of 'valid
-   result hashes' in the image result database file,
-   ``tests/results/imagerepo.json``.
-#. The tests must now be re-run, and the 'new' result should be accepted.
-   Occasionally there are several graphics checks in a single test, only the
-   first of which will be run should it fail.  If this is the case, then you
-   may well encounter further graphical test failures in your next runs, and
-   you must repeat the process until all the graphical tests pass.
-#. To add your changes to Iris, you need to make two pull requests.  The first
-   should be made to the test-images-scitools repository, and this should
-   contain all the newly-generated png files copied into the folder named
-   'image_files'.
-#. The second pull request should be created in the Iris repository, and should
-   only include the change to the image results database
-   (``tests/results/imagerepo.json``) :
-   This pull request must contain a reference to the matching one in
-   test-images-scitools.
+ * The hashes of 'known accepted results' for each test are stored in a
+   lookup dictionary, saved to the repo file
+   ``lib/iris/tests/results/imagerepo.json`` .
+ * An actual reference image for each hash value is stored in a *separate*
+   public repository : https://github.com/SciTools/test-iris-imagehash .
+ * The reference images allow human-eye assessment of whether a new output is
+   judged to be 'close enough' to the older ones, or not.
+ * The utility script ``iris/tests/idiff.py`` automates checking, enabling the
+   developer to easily compare proposed new 'correct' result images against the
+   existing accepted reference images, for each failing test.
 
-Note: the Iris pull-request will not test out successfully in Travis until the
-test-images-scitools pull request has been merged :  This is because there is
-an Iris test which ensures the existence of the reference images (uris) for all
-the targets in the image results database.
 
+How to Add New Known-Valid Result Images
+========================================
+
+When you find that a graphics test in the Iris testing suite has failed,
```

Comment:
Does this process also work for a completely new test? Feels like the subtitle says so but paragraph seems to be exclusively for failures.

Reply:
The title is supposed to indicate that is about adding new "acceptable" result graphics for an existing testcase.  Which you will only do when the (existing) test fails for some reason.
Writing a whole *new* graphics testcase is somewhat trickier.  I decided not to explain that, after all, as we really don't want any more graphics tests unless absolutely necessary !  
It should probably only be done by people smart enough to work it out for themselves 😈

Reply:
By all means suggest improvements to the words so it's clearer ...
? Please ?

Reply:
Like the logic, I'll have a go at some appropriate language. Stand by...

Reply:
See new review